Setup Servers Section

Africk-Hunter opened the pull request:
- Added a section for server setup information called 'Servers'. It might be better to be named something like 'Server Setup' if we are going to eventually have a tab that shows links to various different active KZ servers such as rc noob club; otherwise I think that 'Servers' is probably fine.

- Added a page for setting up a LAN server based on the FKZ docs. I added more detailed information to make it as easy as possible for first timers (as it was my first time doing it and there were points I was confused which I then clarified). I walked through the docs and set up my own server using them to ensure it is correct.

Africk-Hunter committed:
Setup Servers Section

- Added a section for server setup information called 'Servers'. It might be better to be named something like 'Server Setup' if we are going to eventually have a tab that shows links to various different active KZ servers such as rc noob club; otherwise I think that 'Servers' is probably fine.
- Added a page for setting up a LAN server based on the FKZ docs. I added more detailed information to make it as easy as possible for first timers (as it was my first time doing it and there were points I was confused which I then clarified). I walked through the docs and set up my own server using them to ensure it is correct.

Sikarii commented:
Reopening this PR to hopefully fix CI, sorry!

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+<button onclick="document.getElementById('gameinfo-image').classList.toggle('shown')" class='gameinfoButton'>Example Image</button>
+
+<img id="gameinfo-image" class='hidden' src="/images/gameinfo.png">
```
A good alternative to this could be something like below

````suggestion
::: details When complete, the file should look like this:
```txt{22}
"GameInfo"
{
	// ********************************************************************************
	// ********************************************************************************
	// ********************************************************************************
	// DO NOT EDIT THIS FILE DIRECTLY - YOU PROBABLY WANT TO EDIT CSGO_CORE/GAMEINFO.GI
	// ********************************************************************************
	// ********************************************************************************
	// ********************************************************************************

	game 		    "Counter-Strike 2"
	title 		    "Counter-Strike 2"

    LayeredOnMod	csgo_imported

    FileSystem
    {
        SearchPaths
        {
            Game_LowViolence	csgo_lv // Perfect World content override

            Game    csgo/addons/metamod
            Game	csgo
			Game	csgo_imported
			Game	csgo_core
			Game	core

			Mod		csgo
			Mod		csgo_imported
			Mod		csgo_core
```
:::
````

Sikarii commented on .vitepress/theme/style.css:
```diff
@@ -137,3 +137,56 @@
+  border-radius: 4px;
+  padding: 5px;
+  background-color: white;
+}
```
With other suggested changes in this rveiew, this does not need to exist.

```suggestion
```

Sikarii commented on docs/public/images/gameinfo.png:
Delete along with the suggestion in the markdown file.

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+<br>
+
+> [!WARNING]
+> This guide is for **ONLY** for windows devices
```
```suggestion
> This guide is for **ONLY** for Windows devices
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+# Setting Up A LAN Server
```
```suggestion
# Setting up a LAN server
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+> [!WARNING]
+> This guide is for **ONLY** for windows devices
+
+## Server Setup
```
```suggestion
## Server setup
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+2. Inside of ``cs2kz_server`` create another folder called ``steamcmd``
+
+>[!DANGER]
+> Note that the folder which will be used for the server is **REQURIED** to be on a drive with a least
```
```suggestion
> Note that the folder which will be used for the server is **REQUIRED** to be on a drive with a least
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+1. [Server Setup](#server-setup): Setting up the actual server itself.
+2. [Metamod Setup](#metamod-setup): Installing and setting up Metamod: Source.
+3. [KZ Plugins](#kz-plugins): Adding essential plugins to make KZ work properly.
```
```suggestion
1. [Server setup](#server-setup): Setting up the actual server itself.
2. [Metamod setup](#metamod-setup): Installing and setting up Metamod: Source.
3. [KZ plugins](#kz-plugins): Adding essential plugins to make KZ work properly.
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+## Server Setup
+
+### 1. Folder Preparation
```
```suggestion
### 1. Preparation
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+1. Navigate to the [SteamCMD Download Page](https://developer.valvesoftware.com/wiki/SteamCMD#Downloading_SteamCMD).
+
+2. Under step ``[1]`` click the download link to download the ``zip`` file.
```
```suggestion
2. Under step ``[1]`` click the download link to download the zip file.
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+2. Under step ``[1]`` click the download link to download the ``zip`` file.
+
+3. Extract the contents of the ``zip`` into the ``steamcmd`` folder.
```
```suggestion
3. Extract the contents of the zip file into the ``steamcmd`` folder.
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+2. Once it's done, type ``login anonymous`` in the terminal that was opened and press enter.
+
+3. Next type in ``app_update 730 validate`` and press enter. This will begin the installation of the server and may take a very long time
+depending on your internet and drive speeds.
```
```suggestion
3. Next type in ``app_update 730 validate`` and press enter. This will begin the installation of the server and may take a very long time depending on your internet and drive speeds.
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+3. Keep this window open as we will need the token for the next step.
+
+### 5. Running the Server
```
```suggestion
### 5. Running the server
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+>[!DANGER]
+> Do **NOT** close the terminal until it outputs ``Success! App '730' fully installed.``!
+
+### 4. Adding a Steam GSLT Token
```
```suggestion
### 4. Adding a Steam GSLT token
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+
+
+### 3. Installing the Server
```
```suggestion
### 3. Installing the server
```

Sikarii commented on docs/servers/lan.md:
````diff
@@ -0,0 +1,173 @@
+
+2. Right click ``start.bat``, click ``Edit``, and add this text into it:
+
+```txt
````
```suggestion
```batch
```

Sikarii commented on docs/servers/lan.md:
````diff
@@ -0,0 +1,173 @@
+start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount YOUR_TOKEN_HERE
+```
+> [!WARNING]
+> Ensure that you replace ``YOUR_TOKEN_HERE`` with your Steam GSLT Token.
````
```suggestion
> Ensure that you replace ``YOUR_TOKEN_HERE`` with your Steam GSLT token.
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+3. You should now be able save the file, close the text editor, and run it by double-clicking on it. It should run a terminal.
+If you get a prompt to give CS2 access through your firewall, press yes.
+
+### 6. Connecting to the Server
```
```suggestion
### 6. Connecting to the server
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+OR
+
+2. Opening the in-game console and typing in ``connect localhost``
```
```suggestion
You can connect to the server by either:
- Opening the in-game console and typing in ``connect localhost``
- Opening the in-game Community Server Browser and navigating to the ``LAN`` tab, and pressing connect on the server
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+2. Opening the in-game console and typing in ``connect localhost``
+
+## Metamod Setup
```
```suggestion
## Metamod setup
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+### Installing Metamod: Source
+
+1. Open [Metamod: Source (Dev 2.0x) Downloads](https://www.sourcemm.net/downloads.php/?branch=master) and download the
+latest Build for your operating system (Windows/Linux).
```
```suggestion
latest build for your operating system (Windows/Linux).
```

Sikarii commented on docs/servers/lan.md:
````diff
@@ -0,0 +1,173 @@
+Game    csgo/addons/metamod
+```
+to the top of the section with similar inputs (see image below), and save the file.<br><br>
+When complete, it should look like this:
````
Should avoid giving instructions after giving the user input.
````suggestion
4. Ignore any warnings about editing the file, and add the following to the top of the section with similar inputs and save the file:
```txt
Game    csgo/addons/metamod
```
````

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+<img id="gameinfo-image" class='hidden' src="/images/gameinfo.png">
+
+## KZ Plugins
```
```suggestion
## KZ plugins
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+## KZ Plugins
+
+### Installing the KZ Plugin
```
```suggestion
### Installing the KZ plugin
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+2. Extract the ``zip`` (Windows) or ``tar.gz`` (Linux), then drag the ``addons`` and ``cfg`` folders into the ``\csgo\`` folder (The same folder used in the [Metamod Setup](#metamod-setup) step).
+
+### Installing Optional Plugins
```
```suggestion
### Installing optional plugins
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+1. Open [cs2kz-metamod releases](https://github.com/KZGlobalTeam/cs2kz-metamod/releases) and download the latest ``(pre-) release`` for your operating system.
+
+2. Extract the ``zip`` (Windows) or ``tar.gz`` (Linux), then drag the ``addons`` and ``cfg`` folders into the ``\csgo\`` folder (The same folder used in the [Metamod Setup](#metamod-setup) step).
```
```suggestion
2. Extract the ``zip`` (Windows) or ``tar.gz`` (Linux), then drag the ``addons`` and ``cfg`` folders into the ``\csgo\`` folder (The same folder used in the [Metamod setup](#metamod-setup) step).
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+#### Jumpstat/Quake sounds support
+
+1. Open [MultiAddonManager](https://github.com/Source2ZE/MultiAddonManager/releases) Releases and download the latest release for your operating system (Windows/Linux).
```
```suggestion
1. Open [MultiAddonManager](https://github.com/Source2ZE/MultiAddonManager/releases) releases and download the latest release for your operating system (Windows/Linux).
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+2. Extract the ``zip`` (Windows) or ``tar.gz`` (Linux), then drag the ``addons`` and ``cfg`` folders into the ``\csgo\`` folder.
+
+#### Database Support
```
```suggestion
#### Database support
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+2. Extract the ``zip`` (Windows) or ``tar.gz`` (Linux), then drag the ``addons`` and ``cfg`` folders into the ``\csgo\`` folder.
+
+#### Language Support
```
```suggestion
#### Language support
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+#### Language Support
+
+1. Open [GetClientCvarValue's](https://github.com/komashchenko/ClientCvarValue/releases) Releases and download the latest release for your operating system (Windows/Linux).
```
```suggestion
1. Open [GetClientCvarValue](https://github.com/komashchenko/ClientCvarValue/releases) releases and download the latest release for your operating system (Windows/Linux).
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+#### Database Support
+
+1. Open [SQLMM's Releases](https://github.com/zer0k-z/sql_mm/releases) and download the latest release for your operating system (Windows/Linux).
```
```suggestion
1. Open [SQLMM](https://github.com/zer0k-z/sql_mm/releases) releases and download the latest release for your operating system (Windows/Linux).
```

Sikarii commented on docs/servers/lan.md:
````diff
@@ -0,0 +1,173 @@
+2. Edit the ``start.bat`` (created in [Server Setup](#server-setup)) and add either ``+host_workshop_map`` or ``+host_workshop_collection`` command along with the
+``ID`` of the map/collection to the 2nd row on the file. If you've followed this guide exactly as instructed it should look something like this:
+
+```AddingMap
+cd ".\steamcmd\steamapps\common\Counter-Strike Global Offensive\game\bin\win64\"
+start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount GSLT_TOKEN +host_workshop_map MAP_ID
+```
+
+or
+
+```AddingCollection
+cd ".\steamcmd\steamapps\common\Counter-Strike Global Offensive\game\bin\win64\"
+start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount GSLT_TOKEN +host_workshop_collection COLLECTION_ID
+```
````
````suggestion
::: code-group

```batch [Map]
cd ".\steamcmd\steamapps\common\Counter-Strike Global Offensive\game\bin\win64\"
start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount GSLT_TOKEN +host_workshop_map MAP_ID
```

```batch [Collection]
cd ".\steamcmd\steamapps\common\Counter-Strike Global Offensive\game\bin\win64\"
start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount GSLT_TOKEN +host_workshop_collection COLLECTION_ID
```

:::
````

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+
+Replacing ``MAP_ID``/``COLLECTION_ID`` with the correct ID & ``GSLT_TOKEN`` with your token.
+
+> [!INFO]
+> Although maps can be manually added into files, ``FastDL`` does not exist for CS2, so using workshop instead is recommended.
```
```suggestion
> [!INFO]
> Although maps can be manually added into files, FastDL does not exist for CS2, so using Steam Workshop instead is recommended.
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,173 @@
+> [!INFO]
+> Although maps can be manually added into files, ``FastDL`` does not exist for CS2, so using workshop instead is recommended.
+
+> [!WARNING]
+> Currently the CS2 Workshop doesn't require you to use a GSLT token for maps, but this could change in the future.
```
```suggestion
> [!WARNING]
> Currently the Steam Workshop doesn't require you to use an api key for CS2 maps, but this could change in the future.
```

Sikarii reviewed (changes requested):
Additionally, any lines that use single-line codeblocks should use a single backtick opposed to double.

```diff
- ``Thing``
+ `Thing`
```

Africk-Hunter committed:
Applied review changes

I applied all the review changes. Pretty sure I didn't miss anything but let me know if I did because there were a lot of minor changes to make.

Africk-Hunter committed:
One missed change

Co-authored-by: Sikari <37634135+Sikarii@users.noreply.github.com>

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,196 @@
+3. [KZ plugins](#kz-plugins): Adding essential plugins to make KZ work properly.
+4. [Maps](#adding-maps): Adding KZ maps to the server.
+
+<br>
```
```suggestion
```

Sikarii commented on docs/servers/lan.md:
````diff
@@ -0,0 +1,196 @@
+
+> [!INFO]  
+> The `\csgo\` folder can be found in:
+> ```FilePath
````
```suggestion
> ```batch
```

Sikarii commented on docs/servers/lan.md:
````diff
@@ -0,0 +1,196 @@
+```
+
+::: details When complete, the file should look like this:
+```txt
````
Highlight the line that was supposed to be added.

```suggestion
```txt{22}
```

Sikarii commented on docs/servers/lan.md:
```diff
@@ -0,0 +1,196 @@
+
+### 4. Adding a Steam GSLT token
+
+1. Navigate to [Steam GSLT Tokens](https://steamcommunity.com/dev/managegameservers) (log in with steam if you aren't already).
```
```suggestion
1. Navigate to [Steam GSLT tokens](https://steamcommunity.com/dev/managegameservers) (log in with Steam if you aren't already).
```

Sikarii commented on docs/servers/lan.md:
````diff
@@ -0,0 +1,196 @@
+2. Edit the `start.bat` (created in [Server Setup](#server-setup)) and add either `+host_workshop_map` or `+host_workshop_collection` command along with the
+`ID` of the map/collection to the 2nd row on the file. If you've followed this guide exactly as instructed it should look something like this:
+
+::: code-group  
+```batch [Map]
+cd ".\steamcmd\steamapps\common\Counter-Strike Global Offensive\game\bin\win64"
+start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount GSLT_TOKEN +host_workshop_map MAP_ID
+:::
+
+::: code-group
+```batch [Collection]  
+cd ".\steamcmd\steamapps\common\Counter-Strike Global Offensive\game\bin\win64\"  
+start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount GSLT_TOKEN +host_workshop_collection COLLECTION_ID  
+:::
````
````suggestion
::: code-group

```batch [Map]
cd ".\steamcmd\steamapps\common\Counter-Strike Global Offensive\game\bin\win64\"
start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount GSLT_TOKEN +host_workshop_map MAP_ID
```

```batch [Collection]  
cd ".\steamcmd\steamapps\common\Counter-Strike Global Offensive\game\bin\win64\"  
start cs2.exe -dedicated +map de_dust2 +sv_setsteamaccount GSLT_TOKEN +host_workshop_collection COLLECTION_ID
```

:::
````

Sikarii reviewed (changes requested):
Couple things that were missed, otherwise looks good to go!

Africk-Hunter committed:
Resolved Review Changes
